src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -277,12 +277,20 @@ async def get_metric_value(
         self, start_interval: datetime, end_interval: datetime, connection: asyncpg.connection.Connection
     ) -> Sequence[MetricValue]:
         query: str = f"""
-            WITH {LATEST_RELEASED_RESOURCES_SUBQUERY}
-            SELECT r.environment, r.status, COUNT(*)
-            FROM latest_released_resources AS r
-            GROUP BY r.environment, r.status
-            ORDER BY r.environment, r.status
-        """
+                WITH {LATEST_RELEASED_MODELS_SUBQUERY}, nonzero_statuses AS (
+                    SELECT r.environment, r.status, COUNT(*) AS count
+                    FROM {Resource.table_name()} AS r
+                    INNER JOIN latest_released_models AS cm
+                    ON r.environment = cm.environment AND r.model = cm.version
```

Comment:
Sorry about the conflict. I think we can keep a bit more of my changes though, and get the best of both worlds.
```suggestion
                WITH {LATEST_RELEASED_RESOURCES_SUBQUERY}, nonzero_statuses AS (
                    SELECT r.environment, r.status, COUNT(*) AS count
                    FROM latest_released_resources AS r
```